dlup/annotations_experimental.py

```diff
@@ -0,0 +1,1319 @@
+        Returns
+        -------
+        None
+        """
+        self._layers.simplify(tolerance)
+
+    def __contains__(self, item: str | Point | Polygon) -> bool:
+        if isinstance(item, str):
+            return item in self.available_classes
+        if isinstance(item, Point):
+            return item in self._layers.points
+        if isinstance(item, Polygon):
+            return item in self._layers.polygons
+
+        return False
+
+    def __len__(self) -> int:
+        return self._layers.size()
+
+    @property
+    def available_classes(self) -> set[str]:
+        """Get the available classes in the annotations.
+
+        Returns
+        -------
+        set[str]
+            The available classes in the annotations.
+
+        """
+        available_classes = set()
+        for polygon in self._layers.polygons:
+            if polygon.label is not None:
+                available_classes.add(polygon.label)
+        for point in self._layers.points:
+            if point.label is not None:
+                available_classes.add(point.label)
+
+        return available_classes
+
+    def __iter__(self) -> Iterable[Polygon | Point]:
+        # First returns all the polygons then all points
+        for polygon in self._layers.polygons:
+            yield polygon
+
+        for point in self._layers.points:
+            yield point
+
+    def __add__(self, other: Any) -> "SlideAnnotations":
+        """
+        Add two annotations together. This will return a new `SlideAnnotations` object with the annotations combined.
+        The polygons will be added from left to right followed the points from left to right.
+
+        Notes
+        -----
+        - The polygons and points are shared between the objects. This means that if you modify the polygons or points
+          in the new object, the original objects will also be modified. If you wish to avoid this, you must add two
+          copies together.
+        - Note that the sorting is not applied to this object. You can apply this by calling `sort_polygons()` on
+        the resulting object.
+
+        Parameters
+        ----------
+        other : SlideAnnotations
+            The other annotations to add.
+
+        """
+        if not isinstance(other, (SlideAnnotations, Point, Polygon, list)):
+            raise TypeError(f"Unsupported type {type(other)}")
+
+        if isinstance(other, SlideAnnotations):
+            if not self.sorting == other.sorting:
+                raise TypeError("Cannot add annotations with different sorting.")
+            if self.offset_function != other.offset_function:
+                raise TypeError(
+                    "Cannot add annotations with different requirements for offsetting to slide bounds "
+                    "(`offset_function`)."
+                )
+
+            tags: tuple[SlideTag, ...] = ()
+            if self.tags is not None and other.tags is not None:
+                tags = self.tags + other.tags
```

Comment:
This should also consider the case where one is none and the other is defined

Reply:
Fair.